### PR TITLE
add AMS Cluster Info endpoint 

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description
The new endpoint only communicates with AMSAPI, it will be used whenever frontend is unable to retrieve the data from the API otherwise. Some seemingly unnecessary conditions were added because of unexpected behaviour of the API...

Fixes CCXDEV-7682

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Unit tests (no changes in the code)
- Documentation update

## Testing steps
final version tested against production API

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [x] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: amsclient/amsclient.go [hunk 1]
```diff
@@ -170,6 +175,34 @@ func (c *amsClientImpl) GetClusterDetailsFromExternalClusterID(externalID types.
...
 }
 
+func (c *amsClientImpl) GetSingleClusterInfoForOrganization(orgID types.OrgID, clusterID types.ClusterName) (
+	clusterInfo types.ClusterInfo, err error,
+) {
+	tStart := time.Now()
+
+	internalOrgID, err := c.GetInternalOrgIDFromExternal(orgID)
+	if err != nil {
+		return
```
Comment (dpensi) [inline amsclient/amsclient.go:185]: return err ?

File: amsclient/amsclient.go [hunk 2]
```diff
@@ -170,6 +175,34 @@ func (c *amsClientImpl) GetClusterDetailsFromExternalClusterID(externalID types.
...
+	}
+
+	searchQuery := fmt.Sprintf("organization_id = '%s' and external_cluster_id = '%s'", internalOrgID, clusterID)
+
+	subscriptionListRequest := c.connection.AccountsMgmt().V1().Subscriptions().List()
+	clusterInfoList, err := c.executeSubscriptionListRequest(subscriptionListRequest, searchQuery)
+	if err != nil {
+		log.Error().Err(err).Str(clusterIDTag, string(clusterID)).Msg(subscriptionListRequestError)
+		return
```
Comment (dpensi) [inline amsclient/amsclient.go:194]: like above

**Comment by dpensi** [comment]
LGTM

**Review by tisnik — APPROVED** [review]
LGTM

**Pull request merged**